src/dfcx_scrapi/builders/agents.py

```diff
@@ -0,0 +1,254 @@
+"""A set of builder methods to create CX proto resource objects"""
```

Comment:
@MRyderOC awesome work on this! Super easy to use.

One thing we'll need to now do is update `core/agents.py`, specifically the [create_agent](https://github.com/GoogleCloudPlatform/dfcx-scrapi/blob/main/src/dfcx_scrapi/core/agents.py#L217) method.  
This method was written long ago as a precursor to your builders methods, and is much more rudimentary.

Because of that, it's not honoring the full `proto.obj` that we pass in.  
Easy fix though!